File: src/main/java/dev/paigewatson/layoutmaster/data/CarTypeDAL.java
package dev.paigewatson.layoutmaster.data;

import dev.paigewatson.layoutmaster.models.goods.GoodsType;
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation;
import dev.paigewatson.layoutmaster.models.rollingstock.CarType;

import java.util.List;

public interface CarTypeDAL
{
    List<CarType> findAllByCarTypesThatCanCarry(GoodsType expectedGoods);

    CarType findByAarType(AARDesignation aarDesignation);

    void deleteAll();

    CarType insertCarType(CarType carTypeToSave);

    void delete(CarType carTypeToDelete);

    List<CarType> getAllCarTypes();

    List<AARDesignation> getAllAARDesignations();
}

File: src/main/kotlin/dev/paigewatson/layoutmaster/data/CarTypeDAL.kt
package dev.paigewatson.layoutmaster.data

import dev.paigewatson.layoutmaster.models.goods.GoodsType
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation
import dev.paigewatson.layoutmaster.models.rollingstock.CarType

interface CarTypeDAL {
    fun findAllByCarTypesThatCanCarry(expectedGoods: GoodsType): List<CarType>
    fun findByAarType(aarDesignation: AARDesignation): CarType
    fun deleteAll()
    fun insertCarType(carTypeToSave: CarType): CarType
    fun delete(carTypeToDelete: CarType)
    val allCarTypes: List<CarType>
    val allAARDesignations: List<AARDesignation>
}
